ush/test_retrieve_data.py

```diff
@@ -30,78 +34,130 @@ class FunctionalTesting(unittest.TestCase):
 
-            # fmt: off
-            args = [
-                '--file_set', 'fcst',
-                '--config', self.config,
-                '--cycle_date', '2022060112',
-                '--data_stores', 'hpss',
-                '--external_model', 'FV3GFS',
-                '--fcst_hrs', '24', '48', '24',
-                '--output_path', tmp_dir,
-                '--ics_or_lbcs', 'LBCS',
-                '--debug',
-                '--file_type', 'netcdf',
-            ]
-            # fmt: on
+                # fmt: off
+                args = [
+                    '--file_set', 'fcst',
+                    '--config', self.config,
+                    '--cycle_date', date,
+                    '--data_stores', 'hpss',
+                    '--external_model', 'FV3GFS',
+                    '--fcst_hrs', '24',
+                    '--output_path', tmp_dir,
+                    '--ics_or_lbcs', 'LBCS',
+                    '--debug',
+                    '--file_type', 'nemsio',
+                ]
+                # fmt: on
 
-            retrieve_data.main(args)
+                retrieve_data.main(args)
 
-            # Verify files exist in temp dir
+                # Verify files exist in temp dir
 
-            path = os.path.join(tmp_dir, "*")
-            files_on_disk = glob.glob(path)
-            self.assertEqual(len(files_on_disk), 2)
+                path = os.path.join(tmp_dir, "*")
+                files_on_disk = glob.glob(path)
+                self.assertEqual(len(files_on_disk), 1)
+
+    @unittest.skipIf(os.environ.get("CI") == "true", "Skipping HPSS tests")
+    def test_fv3gfs_netcdf_lbcs_from_hpss(self):
+
+        """Get FV3GFS netcdf files from HPSS for LBCS. Tests fcst lead
+        times > 40 hours, since they come from a different archive file.
+        """
+
+        for date in self.dates["FV3GFSgrib2"]:
+            with tempfile.TemporaryDirectory(dir=".") as tmp_dir:
+
+                # fmt: off
+                args = [
+                    '--file_set', 'fcst',
+                    '--config', self.config,
+                    '--cycle_date', '2022060112',
+                    '--data_stores', 'hpss',
+                    '--external_model', 'FV3GFS',
+                    '--fcst_hrs', '24', '48', '24',
+                    '--output_path', tmp_dir,
+                    '--ics_or_lbcs', 'LBCS',
+                    '--debug',
+                    '--file_type', 'netcdf',
+                ]
+                # fmt: on
+
+                retrieve_data.main(args)
+
+                # Verify files exist in temp dir
+
+                path = os.path.join(tmp_dir, "*")
+                files_on_disk = glob.glob(path)
+                self.assertEqual(len(files_on_disk), 2)
 
     # GDAS Tests
     def test_gdas_ics_from_aws(self):
 
         """In real time, GDAS is used for LBCS with a 6 hour offset."""
 
         with tempfile.TemporaryDirectory(dir=".") as tmp_dir:
-            os.chdir(tmp_dir)
```

Comment:
I don't think this should be remove. In NCO mode the locations were conflicting, so I had to use temp directory for it.
This is the commit that changed it
https://github.com/danielabdi-noaa/ufs-srweather-app/commit/f3b1b3f75a051a07d06f59f6caccfd8f4e277705

Reply:
Sorry I don't understand, can you clarify what NCO mode means specifically for these tests? Shouldn't the unit tests be run the same way regardless of NCO or community mode? Here the tmpdir is still used as the destination of the test data, we just don't use that as a working directory.

If the "os.chdir" lines are restored, we will need to include a "chdir .." at the end to avoid this problem of recursively nesting tmpdirs.

Reply:
Cleanup that was not working for NCO mode at the time, and `clean_up_output_dir` still uses `os.getcwd()`. The current path should be EXPTDIR/CDATE for community mode and COMIN for NCO mode for this to work properly. The ex-scripts that call `retrieve_data.py` switch to those directories first, so `test_retrieve_data.py` should do the same and do `chdir ..` later to fix the recursive issue you pointed out.

Reply:
@danielabdi-noaa I have restored the usage of the tmpdir in the unit tests (using an absolute path to avoid previous issues). I will point out that this is not a requirement on the retrieve_data.py script, so theoretically that's something that should be covered in a test of the exscripts, not this unit test. But we can keep it here while we do not have a test specific to the exscripts.